### PR TITLE
Update project

### PR DESCRIPTION
**Pull request by davidbrota**
Update Go to 1.16
Migrate to Go Modules
Update to Infra SDK v3.6.7
Update other dependencies
Add golangci lint to the prerelease pipeline

### CODE REVIEW

File: .github/workflows/push_pr.yml [hunk 1]
```diff
@@ -30,6 +30,13 @@ jobs:
       - name: Validate code
         run: make ci/validate
 
+  golangci-lint:
```
Comment (gsanchezgavier) [inline .github/workflows/push_pr.yml:33]: oh i didn't know we are adding this linter. Do we keep ci/validate?

Comment (davidbrota) [reply .github/workflows/push_pr.yml:18]: I don't think so. I just removed it

File: .gitignore [hunk 2]
```diff
@@ -29,8 +29,7 @@ coverage.xml
 target/
 bin
 tmp/
-go.mod
-go.sum
+vendor/
```
Comment (roobre) [inline .gitignore:32]: What's the motivation to ignore `vendor`? I'd prefer git to complain if it exists.

File: Makefile [hunk 3]
```diff
@@ -3,86 +3,51 @@ export PATH := $(PATH):$(GOPATH)/bin
...
-	@rm -rfv bin coverage.xml
+	@echo "=== $(INTEGRATION) === [ clean ]: cleaning workspace..."
+	@rm -rfv bin coverage.xml vendor
 
-validate-deps:
-	@echo "=== $(INTEGRATION) === [ validate-deps ]: installing validation dependencies..."
-	@go get -v $(VALIDATE_DEPS)
+vendor/modules.txt:
+	@echo "=== $(INTEGRATION) === [ dependencies ]: downloading vendor dependencies..."
+	@go mod vendor
```
Comment (roobre) [inline Makefile:24]: `go build` should already take care of this. Why do we have a target to generate a vendor folder?

File: build/Dockerfile [hunk 4]
```diff
@@ -1,6 +1,6 @@
-FROM golang:1.15.2-buster
+FROM golang:1.16
 
-ARG GH_VERSION='1.1.0'
+ARG GH_VERSION='1.6.0'
```
Comment (roobre) [inline build/Dockerfile:3]:
```suggestion
ARG GH_VERSION='1.9.0'
```

**Comment by roobre** [comment]
Remember to remove the
```
&& go get golang.org/dl/go1.9.7 \
&& /go/bin/go1.9.7 download`
```
lines from Dockerfile

File: Makefile (hunk 3 above)
Comment (davidbrota) [reply Makefile:24]: I was trying to keep it explicit, but you're right.

File: .gitignore (hunk 2 above)
Comment (davidbrota) [reply .gitignore:32]: what would be the use case there? the vendor folder should always be created from the go mod file. I don't see a reason to remove the folder before committing some changes.

File: Makefile (hunk 3 above)
Comment (ivancorrales) [reply Makefile]: I'm not sure if the command `go build` will download the dependencies in the  folder `tools`...   The go build command seems to check only  files (*.go)  in the src folder.

File: .github/workflows/prerelease.yml [hunk 5]
```diff
@@ -9,25 +9,20 @@ on:
...
     runs-on: ubuntu-20.04
     steps:
       - uses: actions/checkout@v2
-      - name: Login to DockerHub
-        uses: docker/login-action@v1
-        with:
-          username: ${{ secrets.OHAI_DOCKER_HUB_ID }}
-          password: ${{ secrets.OHAI_DOCKER_HUB_PASSWORD }}
-      - name: Validate code
-        run: make ci/validate
```
Comment (gsanchezgavier) [inline .github/workflows/prerelease.yml:30]: i understood we were going to keep this and tackle it in a feature iteration

File: Makefile [hunk 6]
```diff
@@ -3,86 +3,47 @@ export PATH := $(PATH):$(GOPATH)/bin
...
-	@echo "=== $(INTEGRATION) === [ test-deps ]: installing testing dependencies..."
-	@go get -v $(TEST_DEPS)
-
-test-only:
+test:
 	@echo "=== $(INTEGRATION) === [ test ]: running unit tests..."
-	@gocov test ./... | gocov-xml > coverage.xml
-
-test: test-deps test-only
+	@go run $(GOFLAGS) $(GOCOV) test ./... | go run $(GOFLAGS) $(GOCOV_XML) > coverage.xml
```
Comment (gsanchezgavier) [inline Makefile:38]: just for curiosity why is the `-mod=mod` needed? I guess is related to the new tool using approach

File: README.md [hunk 7]
```diff
@@ -12,7 +12,7 @@ For installation and usage instructions, see our [documentation web site](https:
 
 ## Building
 
-Golang is required to build the integration. We recommend Golang 1.11 or higher.
+Golang is required to build the integration. We recommend Golang 1.16 or higher.
```
Comment (gsanchezgavier) [inline README.md:15]: could you modify also [line 37](https://github.com/newrelic/nri-nginx/pull/69/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L37) referencing to govendor

**Review by gsanchezgavier — APPROVED** [review]
I perhaps understood incorrectly what we are going to do with the ci/validate action. i left a comment there, the rest lgtm

File: Makefile (hunk 6 above)
Comment (davidbrota) [reply Makefile:38]: yes, the go command will ignore the vendor directory and use the go.mod

File: .github/workflows/prerelease.yml (hunk 5 above)
Comment (davidbrota) [reply .github/workflows/prerelease.yml:30]: I did it here, but we could do it along with the static analysis iteration

**Pull request merged**